### ISSUE TITLE
high memory use when idle

### ISSUE THREAD

**Issue by ahl** [unisonweb/unison#1665]
I left `ucm` running on my Mac in a terminal window. After a few days it was using 12GB of memory.

**Comment by aryairani** [comment]
Thanks for the report.  A fix is in the works for some high memory issues, but my gut feeling is that yours is something different.  Can you elaborate any more about your codebase and `ucm` session?

**Comment by ahl** [comment]
The "codebase" such as it is, is literally my first tinkering with Unison.

```
use .base
use io

> Text.fromCharList(Heap.sort(Text.toCharList("balloon")))
> io.openFile(io.FilePath("/Users/ahl/src/scrabble/word.lst"))
```

I should caveat this with: I have no clue what I'm doing with Unison and basically couldn't find any documentation.

**Comment by aryairani** [comment]
Would you attach your `~/.unisonHistory` as a [gist](https://gist.github.com)
too, after redacting any private information.

**Comment by aryairani** [comment]
Would you attach your `~/.unisonHistory` as a [gist](https://gist.github.com) too, after redacting any private information.

**Comment by ahl** [comment]
```
$ cat .unisonHistory 
pwd
pull https://github.com/unisonweb/base:.releases._latest base
$ 
```

**Comment by aryairani** [comment]
Ok thanks.  So you did the `pull` command, and then you saved your program from https://github.com/unisonweb/unison/issues/1665#issuecomment-680953501, and then went away for a few days and came back to 12GB of memory usage?

But yes IMO `ucm` shouldn't occupy more than a couple MB once it's working correctly.

**Comment by ahl** [comment]
@aryairani yes, that's right. I don't know if it's relevant, but I created a program with two mutually recursive functions in an attempt to see what a stack overflow looks like in Unison. I ^C'ed before stepping away.

**Comment by aryairani** [comment]
Hmm, it's conceivable that it somehow continued to run despite the ^C.  I tried leaving a session running for a few days too (not running anything, and also it was Linux, not Mac) but didn't see the leak you did.

Still, terminated tasks should not continue to consume memory.  If you can reproduce it, that would be great; otherwise we'll just keep an eye on it.
